didasm/src/tokens.rs

```diff
@@ -0,0 +1,352 @@
+                Ok(Operand::Direct(expr))
+            } else if let Some(cap) = CHECK_INSIDE_BRACKETS.captures(op) {
+                let op1 = RegOrExpr::from_str(cap.get(1).unwrap().as_str())?;
+                let op2 = RegOrExpr::from_str(cap.get(2).unwrap().as_str())?;
+                let op3 = cap.get(3).map(|x| RegOrExpr::from_str(x.as_str())).transpose()?;
+                let sign = cap.get(4).map(|x| x.as_str());
+                match (op1, op2, op3, sign) {
+                    (Reg( b @ (Ba|Bb) ), Reg( x @ (Xa|Xb) ), None, None) |
+                    (Reg( x @ (Xa|Xb) ), Reg( b @ (Ba|Bb) ), None, None)
+                        => Ok(Operand::RegSum { b, x }),
+                    (Reg( b @ (Ba|Bb) ), Reg( x @ (Xa|Xb) ), Some(Expression(e)), None) |
+                    (Reg( x @ (Xa|Xb) ), Reg( b @ (Ba|Bb) ), Some(Expression(e)), None) |
+                    (Reg( b @ (Ba|Bb) ), Expression(e), Some(Reg( x @ (Xa|Xb) )), None) |
+                    (Reg( x @ (Xa|Xb) ), Expression(e), Some(Reg( b @ (Ba|Bb) )), None) |
+                    (Expression(e), Reg( b @ (Ba|Bb) ), Some(Reg( x @ (Xa|Xb) )), None) |
+                    (Expression(e), Reg( x @ (Xa|Xb) ), Some(Reg( b @ (Ba|Bb) )), None)
+                        => Ok(Operand::BasedIndexed { b, x, depls: e }),
+                    (Reg( b @ (Ba|Bb) ), Reg( x @ (Xa|Xb) ), None, Some("+")) |
+                    (Reg( x @ (Xa|Xb) ), Reg( b @ (Ba|Bb) ), None, Some("+"))
+                        => Ok(Operand::RegSumAutoincrement { b, x }),
+                    (Reg( b @ (Ba|Bb) ), Reg( Xa ), None, Some("-")) |
+                    (Reg( Xa ), Reg( b @ (Ba|Bb) ), None, Some("-"))
+                        => Ok(Operand::RegSumAutodecrement { b }),
+                    (Reg(_), Reg( Xb ), None, Some("-")) |
+                    (Reg( Xb ), Reg(_), None, Some("-"))
+                        => Err(format!("xb is not supported in autodecrement instructions")),
+                    (Reg( b @ (Ba|Bb) ), Expression(e), None, None) |
+                    (Expression(e), Reg( b @ (Ba|Bb) ), None, None) 
+                        => Ok(Operand::Based { b, depls: e }),
+                    (Reg( x @ (Xa|Xb) ), Expression(e), None, None) |
+                    (Expression(e), Reg( x @ (Xa|Xb) ), None, None) 
+                        => Ok(Operand::Indexed { x, depls: e }),
+                    
+    
+
+
+                    (Reg(_), Reg(_), Some(Reg(_)), _)
+                        => Err(format!("Sum between 3 registers is not supported by the ISA")),
+                    (Reg(Ba|Bb), Reg(Ba|Bb), _, _) | (Reg(Xa|Xb), Reg(Xa|Xb), _, _) |
+                    (_, Reg(Ba|Bb), Some(Reg(Ba|Bb)), _) | (_, Reg(Xa|Xb), Some(Reg(Xa|Xb)), _) |
+                    (Reg(Ba|Bb), _, Some(Reg(Ba|Bb)), _) | (Reg(Xa|Xb), _, Some(Reg(Xa|Xb)), _)
+                        => Err(format!("Using the same register group (B/X) between 2 registers is not allowed in an operand")),
```

Comment:
instead of format! use "".to_string()
use format! only when you need to insert variables in strings